### DISCUSSION TITLE
Self-aware instrumentation

### DISCUSSION THREAD

**Discussion by jtayl222** [open-telemetry/opentelemetry-java-instrumentation#8843] [Q&A]
We know that we can choose methods to instrument at JDK launch with agent.properties, but, in a long running enterprise application, this produces more data than we are capable of consuming. Is it possible develop a self-aware application so that, at runtime, the methods that will be instrumented will be switched on and off based on the load-specific details? How can I control instrumentation at runtime?

**Comment by trask** [comment]
you could implement a custom sampler similar to the Jaeger remote sampler (https://github.com/open-telemetry/opentelemetry-java/tree/main/sdk-extensions/jaeger-remote-sampler) which can be controlled at runtime
